app/components/candidate_interface/degree_required_component.rb

```diff
@@ -0,0 +1,73 @@
+class CandidateInterface::DegreeRequiredComponent < ViewComponent::Base
+  CANDIDATE_GRADES = {
+    'Upper second-class honours (2:1)' => 4,
+    'Lower second-class honours (2:2)' => 3,
+    'Third-class honours' => 2,
+    'Pass' => 1,
+  }.freeze
+
+  COURSE_REQUIRED_GRADES = {
+    'two_one' => 4,
+    'two_two' => 3,
+    'third_class' => 2,
+  }.freeze
+
+  COURSE_REQUIRED_GRADE_TEXT = {
+    'two_one' => '2:1 degree or higher (or equivalent)',
+    'two_two' => '2:2 degree or higher (or equivalent)',
+    'third_class' => 'Third-class degree or higher (or equivalent)',
+    'not_required' => 'Any degree grade',
+  }.freeze
+
+  DEGREE_GRADE_INSUFFICIENT_TEXT = {
+    'Lower second-class honours (2:2)' => 'a 2:2 degree.',
+    'Third-class honours' => 'a Third class degree.',
+    'Pass' => 'an Ordinary degree (pass).',
+  }.freeze
+
+  attr_accessor :application_choice, :course_degree_requirements, :uk_bachelor_degrees
+
+  def initialize(application_choice)
+    @application_choice = application_choice
+    @course_degree_requirements = application_choice.current_course.degree_grade
+    @uk_bachelor_degrees = find_uk_bachelor_degrees(application_choice)
+  end
+
+  def candidate_degree_grade_below_required_grade?
+    return false if @uk_bachelor_degrees.empty? || degrees_with_other_grades_only? || bachelors_and_masters_degree?
```

Comment:
I _think_ rather than the `bachelors_and_masters_degree?` condition, we could swap this out with a `has_non_uk_bachelors_degree?` condition?  ie it should return false if:

* candidate has a Masters or PhD, or any type of free-text degree type
* candidate has a non-UK degree (even if they also have a UK bachelors degree)

Reply:
Maybe a bit of clarification here. The `find_uk_bachelor_degrees(application_choice)` is currently gathering all bachelors degrees held by the candidate. It also excludes any nonuk degrees and other type degrees.

`@uk_bachelor_degrees.empty?` condition checks to see if this method has returned any results.

`degrees_with_other_grades_only?` prevents a situation where there are bachelor degrees but with non standard grades given.

The idea behind `bachelors_and_masters_degree?` was based on Tom's comments at dev-design yesterday as we want to not show guidance if a candidate has a bachelors below the required level but also has a masters degree as this may be sufficient in the eyes of a provider. It's a niche bit of behaviour, but  this is the most straightforward way of achieving this I can think of.

Can discuss further irl if you want? ;)

Reply:
@george-bit I was thinking that `bachelors_and_masters_degree?` wouldn't catch other postgraduate types (eg PhD). But that's a super edge case, and they'd likely have also added their Masters anyway...